### ISSUE TITLE
no panel icon on start

### ISSUE THREAD

**Issue by ameyjah** [nvbn/everpad#147]
Hello,

I have installed everpad on my ubuntu 12.04 64bit setup.

Then I tried to start everpad from unity dash but did not see any panel icon.

So I tried to start using terminal and got following message but no panel icon.

xyz@ubuntu:~$ everpad

(python2.7:7381): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(python2.7:7381): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(python2.7:7381): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(python2.7:7381): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

Is there any way to manually add account info or configure everpad or is there any way to solve the above problem.

--Amey

**Comment by mhaggag** [comment]
The warnings come from a known [ubuntu bug](https://bugs.launchpad.net/ubuntu/+source/light-themes/+bug/762167). You should be able to fix them by installing [gtk2-engines-pixbuf](http://packages.ubuntu.com/gtk2-engines-pixbuf). The warnings should not stop the application from launching.

The most likely cause for the problem is not having [sni-qt](https://launchpad.net/sni-qt) installed. This is a Qt plugin that makes sure applications using QSystemTrayIcon (like everpad) properly show an icon in the notification area. (Basically, Ubuntu introduced a new API for creating such icons, and the library sni-qt magically fixes applications to use this new API).

To recap, the following should solve the problems:
sudo apt-get install gtk2-engines-pixbuf
sudo apt-get install sni-qt

If you do have sni-qt installed..well, we'll need to do some further investigation :)

**Comment by ameyjah** [comment]
Hello,

Thanks for prompt reply.

I installed pixbuf but that did not help. Sni-qt was already installed.
Is there any log file that we can check for errors?

--Amey

On Sun, Nov 25, 2012 at 8:05 PM, Muhammad Haggag
notifications@github.comwrote:

> The warnings come from a known ubuntu bughttps://bugs.launchpad.net/ubuntu/+source/light-themes/+bug/762167.
> You should be able to fix them by installing gtk2-engines-pixbufhttp://packages.ubuntu.com/gtk2-engines-pixbuf.
> The warnings should not stop the application from launching.
> 
> The most likely cause for the problem is not having sni-qthttps://launchpad.net/sni-qtinstalled. This is a Qt plugin that makes sure applications using
> QSystemTrayIcon (like everpad) properly show an icon in the notification
> area. (Basically, Ubuntu introduced a new API for creating such icons, and
> the library sni-qt magically fixes applications to use this new API).
> 
> To recap, the following should solve the problems:
> sudo apt-get install gtk2-engines-pixbuf
> sudo apt-get install sni-qt
> 
> If you do have sni-qt installed..well, we'll need to do some further
> investigation :)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/nvbn/everpad/issues/147#issuecomment-10701521.

## 

_Regards,_
_Amey Jahagirdar_
*
*
__

**Comment by mhaggag** [comment]
Not that I know of. So is everpad actually running in the background, or does it terminate? Can you try "ps -e | grep everpad"?

**Comment by ameyjah** [comment]
Yeah I can see
xyzz      2780     1  0 15:36 ?        00:00:00 /usr/bin/python2.7 /usr/bin/everpad-lens
xyzz    2873     1  0 15:36 ?        00:00:00 /usr/bin/python2.7 /usr/bin/everpad-provider
xyzz      9321  9262  2 20:45 pts/2    00:00:00 /usr/bin/python2.7 /usr/bin/everpad

I think it keeps on running from background. Is there any way, I can open this python app as normal window application and not as panel app?

**Comment by mhaggag** [comment]
It's possible to add a mode of operation where everpad doesn't minimize to the notification area, but I think we should really work out why the icon isn't showing up.

I'm preparing a small test app that we can use to debug the indicator icon problem. In the meantime, you can try running everpad as follows:
everpad --settings (to open settings window and set it up)
everpad --create (to create a new note)
everpad --open note (to open the given note)

Unfortunately, there's no command-line argument to get it to show the main window as far as I can see.

**Comment by ameyjah** [comment]
Thank you for assistance.
I could use settings option to authorize my account.

However, I am not able to use --create and --open. I am getting following
error

Traceback (most recent call last):
  File "/usr/bin/everpad", line 9, in <module>
    load_entry_point('everpad==2.2.1dev', 'gui_scripts', 'everpad')()
  File "/usr/lib/pymodules/python2.7/everpad/pad/indicator.py", line 271,
in main
    app.indicator.create()
  File "/usr/lib/pymodules/python2.7/everpad/pad/indicator.py", line 143,
in create
    self.app.provider.create_note(note_struct),
  File "/usr/lib/pymodules/python2.7/everpad/tools.py", line 30, in wrapper
    return getattr(self.**interface, name)(_args, *_kwargs)
  File "/usr/lib/python2.7/dist-packages/dbus/proxies.py", line 70, in
__call**
    return self._proxy_method(_args, *_keywords)
  File "/usr/lib/python2.7/dist-packages/dbus/proxies.py", line 145, in
__call__
    *_keywords)
  File "/usr/lib/python2.7/dist-packages/dbus/connection.py", line 651, in
call_blocking
    message, timeout)
dbus.exceptions.DBusException:
org.freedesktop.DBus.Python.sqlalchemy.orm.exc.NoResultFound: Traceback
(most recent call last):
  File "/usr/lib/python2.7/dist-packages/dbus/service.py", line 707, in
_message_cb
    retval = candidate_method(self, *args, *_keywords)
  File "/usr/lib/pymodules/python2.7/everpad/provider/service.py", line
187, in create_note
    btype.Note.from_tuple(data).give_to_obj(note)
  File "/usr/lib/pymodules/python2.7/everpad/basetypes.py", line 53, in
give_to_obj
    setattr(obj, field[0] + '_dbus', val)
  File "/usr/lib/pymodules/python2.7/everpad/provider/models.py", line 80,
in notebook_dbus
    Notebook.default == True,
  File "/usr/lib/python2.7/dist-packages/sqlalchemy/orm/query.py", line
2020, in one
    raise orm_exc.NoResultFound("No row was found for one()")
NoResultFound: No row was found for one()

xyz@ubuntu:~$ everpad --open 1
Traceback (most recent call last):
  File "/usr/bin/everpad", line 9, in <module>
    load_entry_point('everpad==2.2.1dev', 'gui_scripts', 'everpad')()
  File "/usr/lib/pymodules/python2.7/everpad/pad/indicator.py", line 269,
in main
    app.indicator.open(args.open)
  File "/usr/lib/pymodules/python2.7/everpad/pad/indicator.py", line 117,
in open
    old_note_window = self.opened_notes.get(note.id, None)
AttributeError: 'int' object has no attribute 'id'

Additionally, the app is no syncing the notes from my account. I could not
find note in dash lens.

--Amey

**Comment by mhaggag** [comment]
Hm, not sure what's going on with --create. Probably the main notebook
doesn't exist. Let's focus on the indicator icon problem for now. I'll post
the test app shortly. Can you tell me what you get when you do:
dpkg -l sni-qt

Do you have both amd64 and i386 versions installed? ("ii" at the beginning
of the line)

**Comment by ameyjah** [comment]
Yes. ii at the beginning of the line.

Desired=Unknown/Install/Remove/Purge/Hold
|
Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version        Description
+++-==============-==============-============================================
ii  sni-qt         0.2.5-0ubuntu3 indicator support for Qt

**Comment by mhaggag** [comment]
Can you download and run [this app](http://ubuntuone.com/0c9UmuZ3lLqZdwQmAFfYR8). It's a tiny python application that does nothing but create an indicator icon in the notification area (it should show a save icon). You can run it as:
python testapp.py

**Comment by ameyjah** [comment]
I could not see the icon and got following message.

xyz@ubuntu:~$ python testapp.py
Creating indicator..
Setting icon..
Showing indicator..
QSystemTrayIcon::setVisible: No Icon set
Entering main loop..

--Amey

**Comment by mhaggag** [comment]
You seem to have one version of the library installed. I have two:
    ii  sni-qt:amd64                                  0.2.6-0ubuntu1              amd64                       indicator support for Qt
    ii  sni-qt:i386                                   0.2.6-0ubuntu1              i386                        indicator support for Qt

Also, the error you saw with --create seems to be [this bug](https://github.com/nvbn/everpad/issues/68). I'll download a 12.04 Ubuntu VM and investigate.

**Comment by mhaggag** [comment]
Which theme are you using?

**Comment by ameyjah** [comment]
Default ambiance on ubuntu 12.04

**Comment by ameyjah** [comment]
To update, after restart create works properly and my notes have been
synced correctly in the dash.

Only problem is I can not see icon in panel.

On Mon, Nov 26, 2012 at 10:53 AM, ameyjah@gmail.com ameyjah@gmail.comwrote:

> Default ambiance on ubuntu 12.04

## 

_Regards,_
_Amey Jahagirdar_
*
*
__

**Comment by mhaggag** [comment]
I updated the [test python app](http://ubuntuone.com/0c9UmuZ3lLqZdwQmAFfYR8). Download [this icon](http://ubuntuone.com/40G2HzCmW6YZ9140jvUxnE) alongside it. Do you see an everpad icon when you run it?

**Comment by ameyjah** [comment]
Still no icon.
Got following message.

xyz@ubuntu:~/Downloads$ python testapp.py
Creating indicator..
Setting icon..
Showing indicator..
Entering main loop..

**Comment by mhaggag** [comment]
Alright, that points to a fundamental problem with Qt/Ubuntu system tray icons on Ubuntu 12.04--the test app basically sets up a tray icon and does nothing else. The output you pasted means no errors occurred either. I'll get back to you with further details once I have the 12.04 VM set up.

Thanks for your patience and cooperation so far :)

**Comment by mhaggag** [comment]
I created an amd64 Ubuntu 12.04 VM on VirtualBox. The test app worked fine, and so did everpad. It seems the problem is specific to your system/configuration, somehow. I don't have any clues at the moment, but I'll try to think of something.

Sorry :(

**Comment by ameyjah** [comment]
Thank you for taking time to look into this. I too believe that I must have
messed up my qt configuration somehow. I will try to reinstall qt-packages
if possible and try once again.

**Comment by mhaggag** [comment]
Just to confirm this isn't caused by bad unity settings, what do you get when you run the following command?
gsettings get com.canonical.Unity.Panel systray-whitelist

**Comment by ameyjah** [comment]
gsettings get com.canonical.Unity.Panel systray-whitelist
['JavaEmbeddedFrame', 'Wine', 'Update-notifier', 'desura']

**Comment by mhaggag** [comment]
Can you try adding everpad to that list?

gsettings set com.canonical.Unity.Panel systray-whitelist "[JavaEmbeddedFrame', 'Wine', 'Update-notifier', 'desura', 'everpad']"

You'll have to log out and back in for it to take effect. If everpad starts showing up, then Qt is definitely creating a regular tray icon (rather than an application indicator).

**Comment by StianOby** [comment]
I see the same issue in Ubuntu 12.10.
